src/ctl.c

```diff
@@ -1536,12 +1528,9 @@ thread_arena_ctl(tsd_t *tsd, const size_t *mib, size_t miblen, void *oldp,
 		}
 		/* Set new arena/tcache associations. */
 		arena_migrate(tsd, oldind, newind);
-		if (config_tcache) {
-			tcache_t *tcache = tsd_tcachep_get(tsd);
-			if (tcache != NULL) {
-				tcache_arena_reassociate(tsd_tsdn(tsd), tcache,
-				    newarena);
-			}
+		if (opt_tcache) {
```

Comment:
Ah, it looks to me like the right thing to do is *replace* this check with ```if (tcache_available(tsd)) { ... }```, here, in ```thread_tcache_flush_ctl()```, and the assert in ```tcache_flush()```.  Does that sound right to you?